force-app/main/service/AccountsServiceImpl.cls

```diff
@@ -73,30 +73,34 @@ public inherited sharing class AccountAdapter extends fflib_SObjects2 {
         set;
     }
 
-    public AccountAdapter(List<Account> records) {
+    public AccountsServiceImpl(){}
+    // TODO: SObjects2 should be used by domain classes and not services
+    public AccountsServiceImpl(List<Account> records) {
         this(records, null);
     }
 
-    public AccountAdapter(List<Account> records, List<Account> oldlist) {
+    public AccountsServiceImpl(List<Account> records, List<Account> oldlist) {
         super(records, Account.SObjectType);
         this.oldMap = oldlist == null ? new Map<Id, Account>() : new Map<Id, Account>
                 (oldlist);
     }
 
-    public void onBeforeInsert() {
-        setAccountTypeAndIndividualFieldValues();
-        if (AddressService.isAddressManagementEnabled()) {
-            normalizeLineBreaksFor(householdAccounts());
-        }
-        if (AddressService.isOrgAccountAddressesEnabled()) {
-            normalizeLineBreaksFor(orgAccounts());
-        }
+    public void onBeforeInsert(List<Account> accounts) {
+        setAccountTypeAndIndividualFieldValues(accounts);
+
+        AddressesService.normalizeLineBreaksFor(
+            Households.newInstance(accounts).getRecords()
+        );
+
+        AddressesService.normalizeLineBreaksFor(
```

Comment:
It looks like this method checks for `isAddressManagementEnabled` and not `isOrgAccountAddressesEnabled` now, which is a separate setting. We could vary the check based on the account type being sent?